From 2d50c793b311104bbed269ff14a8a6636547ddee Mon Sep 17 00:00:00 2001
From: Peter Chen <peter.chen@nxp.com>
Date: Thu, 21 Sep 2017 17:08:34 +0800
Subject: [PATCH 2571/5242] MLK-16522-2 usb: cdns3: host: change remove
 sequence for hcd

commit  29ca6d5bec734404f036fab1ac21df4be5c3c3ae from
https://source.codeaurora.org/external/imx/linux-imx.git

At probe, the main hcd is added first, then shared_hcd is added later,
so when we tries to remove hcds, the shared_hcd needs to remove first.

BuildInfo:
- SCFW 1f59442e, IMX-MKIMAGE fb52c576, ATF
- U-Boot 2017.03-imx_v2017.03+g34be5a2

Acked-by: Li Jun <jun.li@nxp.com>
Signed-off-by: Peter Chen <peter.chen@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/usb/cdns3/host.c |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/usb/cdns3/host.c b/drivers/usb/cdns3/host.c
index 621e96f..64bc9cc 100644
--- a/drivers/usb/cdns3/host.c
+++ b/drivers/usb/cdns3/host.c
@@ -172,11 +172,11 @@ static void cdns3_host_stop(struct cdns3 *cdns)
 	if (dev) {
 		hcd = dev_get_drvdata(dev);
 		xhci = hcd_to_xhci(hcd);
-		usb_remove_hcd(hcd);
 		usb_remove_hcd(xhci->shared_hcd);
+		usb_remove_hcd(hcd);
 		synchronize_irq(cdns->irq);
-		usb_put_hcd(hcd);
 		usb_put_hcd(xhci->shared_hcd);
+		usb_put_hcd(hcd);
 		cdns->host_dev = NULL;
 		device_del(dev);
 		put_device(dev);
-- 
1.7.9.5

